core/lib/zksync_core/src/commitment_generator/mod.rs

```diff
@@ -210,11 +214,35 @@ impl CommitmentGenerator {
             }
             state_diffs.sort_unstable_by_key(|rec| (rec.address, rec.key));
 
+            let kzg_settings = load_kzg_settings();
+
+            let blob_commitments = if protocol_version.is_post_1_4_2() {
+                let blob_commitments = header
```

Comment:
consider moving this logic into kzg.rs itself - and having a function 

```
pubdata_to_blob_commitments(Vec<u8>) -> [H256;2].

```

(this way we keep all this logic in one place)